sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -497,6 +497,18 @@ abstract class SchemaPruningSuite
         Row(Row("Janet", null, "Jones"), "Jones") ::Nil)
   }
 
+  testSchemaPruning("SPARK-32163: nested pruning should work even with cosmetic variations") {
+    withTempView("contact_alias") {
+      sql("select * from contacts")
+        .repartition(100, col("name.first"), col("name.last"))
```

Comment:
This issue cannot happen in branch-3.0? 
https://github.com/apache/spark/blob/fc2660c302b0c83a9a8a5bec3cc7ae28f8fecdd6/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala#L80-L85

Reply:
This test case is only for master branch. However this issue can happen in branch-3.0 too. I added another new test here, which is for branch-3.0.

But when we backport this to branch-3.0, we need to remove first test case it will fail on `checkScan(query, "struct<name:struct<first:string,last:string>>")`, because branch-3.0 doesn't prune for repartition by expression.

Reply:
Yea, thanks for adding the test.